src/ImageSharp/Processing/Processors/Dithering/ErrorDither.cs

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) Six Labors and contributors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using SixLabors.ImageSharp.Advanced;
+using SixLabors.ImageSharp.PixelFormats;
+using SixLabors.ImageSharp.Processing.Processors.Quantization;
+
+namespace SixLabors.ImageSharp.Processing.Processors.Dithering
+{
+    /// <summary>
+    /// An error diffusion dithering implementation.
+    /// <see href="http://www.efg2.com/Lab/Library/ImageProcessing/DHALF.TXT"/>
+    /// </summary>
+    public readonly partial struct ErrorDither : IDither, IEquatable<ErrorDither>
```

Comment:
I wonder what are the use-cases for equality comparison. Should we override also `==`? Do we also need to implement `IEquatable<IDither>`? Don't we need a few tests to make sure it works as intended?

While analyzing this I found #1116.

Reply:
Yeah we should implement `IEquatable<IDither>` plus operators. I'll do that now.

Reply:
Done. Also fixed the `DenseMatrix<T>` hashcode generation and added tests.